Index: src/widget/DropdownTimePicker.js
===================================================================
--- src/widget/DropdownTimePicker.js	(revision 8599)
+++ src/widget/DropdownTimePicker.js	(working copy)
@@ -67,6 +67,9 @@
 		// 	name of the form element, used to create a hidden field by this name for form element submission.
 		name: "",
 
+               templatePath: dojo.uri.dojoUri("src/widget/templates/TimePicker.htm"),
+               templateCssPath: dojo.uri.dojoUri("src/widget/templates/TimePicker.css"),
+
 		postMixInProperties: function() {
 			// summary: see dojo.widget.DomWidget
 			dojo.widget.DropdownTimePicker.superclass.postMixInProperties.apply(this, arguments);
@@ -115,7 +118,9 @@
 				value = dojo.date.fromRfc3339(d + "T" + orig);
 			}
 			
-			var tpArgs = { widgetContainerId: this.widgetId, lang: this.lang, value: value };
+                       var tpArgs = { widgetContainerId: this.widgetId, lang: this.lang, value: value,
+                               templateCssPath:  this.templateCssPath, templatePath: this.templatePath
+                               };
 			this.timePicker = dojo.widget.createWidget("TimePicker", tpArgs, this.containerNode, "child");
 			
 			dojo.event.connect(this.timePicker, "onValueChanged", this, "_updateText");
Index: src/widget/DropdownDatePicker.js
===================================================================
--- src/widget/DropdownDatePicker.js	(revision 8599)
+++ src/widget/DropdownDatePicker.js	(working copy)
@@ -93,6 +93,9 @@
 		//	disable all incremental controls, must pick a date in the current display
 		staticDisplay: false,
 		
+               templatePath:  dojo.uri.dojoUri("src/widget/templates/DatePicker.htm"),
+               templateCssPath:  dojo.uri.dojoUri("src/widget/templates/DatePicker.css"),
+
 		postMixInProperties: function(localProperties, frag){
 			// summary: see dojo.widget.DomWidget
 
@@ -120,7 +123,9 @@
 			//attributes to be passed on to DatePicker
 			var dpArgs = {widgetContainerId: this.widgetId, lang: this.lang, value: this.value,
 				startDate: this.startDate, endDate: this.endDate, displayWeeks: this.displayWeeks,
-				weekStartsOn: this.weekStartsOn, adjustWeeks: this.adjustWeeks, staticDisplay: this.staticDisplay};
+                               weekStartsOn: this.weekStartsOn, adjustWeeks: this.adjustWeeks, staticDisplay: this.staticDisplay, 
+                               templateCssPath:  this.templateCssPath, templatePath: this.templatePath
+                               };
 
 			//build the args for DatePicker based on the public attributes of DropdownDatePicker
 			this.datePicker = dojo.widget.createWidget("DatePicker", dpArgs, this.containerNode, "child");
