--- ./FloatingPane.js	2007-05-14 16:52:00.000000000 +0200
+++ src/widget/FloatingPane.js	2007-05-14 16:52:11.000000000 +0200
@@ -421,6 +421,7 @@
 
 		windowState: "minimized",
 		displayCloseAction: true,
+		
 		postCreate: function(){
 			dojo.widget.ModalDialogBase.prototype.postCreate.call(this);
 			dojo.widget.ModalFloatingPane.superclass.postCreate.call(this);
@@ -430,6 +431,8 @@
 			dojo.widget.ModalFloatingPane.superclass.show.apply(this, arguments);
 			//place the background div under this modal pane
 			this.bg.style.zIndex = this.domNode.style.zIndex-1;
+			//Allow to input via keyboard
+			this.giveMeFocus();
 		},
 		hide: function(){
 			this.hideModalDialog();
@@ -439,11 +442,22 @@
 		},
                reOrderWindowsAndBg: function(){
                        var floatingPanes= dojo.widget.manager.getWidgetsByType(this.widgetType);
+			for (var x=0; x<floatingPanes.length; x++) {
+				if (this.widgetId != floatingPanes[x].widgetId) {
+					var floating = floatingPanes[x];
+					floating.bg.style.zIndex = floating.domNode.style.zIndex-1;
+				}
+			}
+			//disconnect on the hidden window, will be set again in show
+			dojo.event.disconnect(document.documentElement, "onkey", this, "_onKey");
+		},
+		giveMeFocus: function(){
+			//disconnect onkey on all window except me
+			var floatingPanes= dojo.widget.manager.getWidgetsByType(this.widgetType);
                        var windows = [];
                        for (var x=0; x<floatingPanes.length; x++) {
                                if (this.widgetId != floatingPanes[x].widgetId) {
-                                               var floating = floatingPanes[x];
-                                               floating.bg.style.zIndex = floating.domNode.style.zIndex-1;
+					dojo.event.disconnect(document.documentElement, "onkey", floatingPanes[x], "_onKey");
                                }
                        }
                },
