Index: src/widget/FloatingPane.js
===================================================================
--- src/widget/FloatingPane.js	(revision 8599)
+++ src/widget/FloatingPane.js	(working copy)
@@ -433,11 +433,26 @@
 		},
 		hide: function(){
 			this.hideModalDialog();
+                       //need also to reinit order between floating and background
+                       this.reOrderWindowsAndBg();
 			dojo.widget.ModalFloatingPane.superclass.hide.apply(this, arguments);
 		},
+               reOrderWindowsAndBg: function(){
+                       var floatingPanes= dojo.widget.manager.getWidgetsByType(this.widgetType);
+                       var windows = [];
+                       for (var x=0; x<floatingPanes.length; x++) {
+                               if (this.widgetId != floatingPanes[x].widgetId) {
+                                               var floating = floatingPanes[x];
+                                               floating.bg.style.zIndex = floating.domNode.style.zIndex-1;
+                               }
+                       }
+               },
 		closeWindow: function(){
 			this.hide();
 			dojo.widget.ModalFloatingPane.superclass.closeWindow.apply(this, arguments);
-		}
+               },
+               onMouseDown : function(){
+                       //Do nothing, in that case we DO NOT need to bring the clicked window to the top 
+               }
 	}
 );
