Index: src/widget/RichText.js
===================================================================
--- src/widget/RichText.js	(revision 8599)
+++ src/widget/RichText.js	(working copy)
@@ -15,7 +15,16 @@
 	(function(){
 		var savetextarea = dojo.doc().createElement('textarea');
 		savetextarea.id = "dojo.widget.RichText.savedContent";
-		savetextarea.style = "display:none;position:absolute;top:-100px;left:-100px;height:3px;width:3px;overflow:hidden;";
+ 		//savetextarea.style = "display:none;position:absolute;top:-100px;left:-100px;height:3px;width:3px;overflow:hidden;";
+ 		with (savetextarea.style) {
+ 			display="none";
+ 			position="absolute";
+ 			top="-100px";
+ 			left="-100px";
+ 			height="3px";
+ 			width="3px";
+ 			overflow="hidden";
+ 		}
 		dojo.body().appendChild(savetextarea);
 	})();
 }else{
