Index: src/widget/TimePicker.js
===================================================================
--- src/widget/TimePicker.js	(revision 8621)
+++ src/widget/TimePicker.js	(working copy)
@@ -7,7 +7,7 @@
 dojo.require("dojo.dom");
 dojo.require("dojo.html.style");
 
-dojo.requireLocalization("dojo.i18n.calendar", "gregorian");
+//dojo.requireLocalization("dojo.i18n.calendar", "gregorian");
 dojo.requireLocalization("dojo.widget", "TimePicker");
 
 dojo.widget.defineWidget(
@@ -55,9 +55,15 @@
 		this.initUI();
 	},
 
-	postMixInProperties: function(localProperties, frag) {
+	postMixInProperties: function(localProperties, frag) { 
 		dojo.widget.TimePicker.superclass.postMixInProperties.apply(this, arguments);
-		this.calendar = dojo.i18n.getLocalization("dojo.i18n.calendar", "gregorian", this.lang); // "am","pm"
+		//Ugly hack : the following line seems to sometimes sucks 
+		//this.calendar = dojo.i18n.getLocalization("dojo.i18n.calendar", "gregorian", this.lang); // "am","pm"
+		this.calendar={};
+		this.calendar.am="AM";
+		this.calendar.pm="PM";
+		this.calendar["field-hour"] = "hour";
+		this.calendar["field-minute"] = "minute";
 		this.widgetStrings = dojo.i18n.getLocalization("dojo.widget", "TimePicker", this.lang); // "any"
 	},
 
