Index: src/widget/DropdownTimePicker.js
===================================================================
--- src/widget/DropdownTimePicker.js	(revision 8620)
+++ src/widget/DropdownTimePicker.js	(working copy)
@@ -41,6 +41,9 @@
 		//	ex: 12:00
 		value: "",
 
+		templatePath: dojo.uri.dojoUri("src/widget/templates/TimePicker.html"),
+		templateCssPath: dojo.uri.dojoUri("src/widget/templates/TimePicker.css"),
+
 		postMixInProperties: function() {
 			dojo.widget.DropdownTimePicker.superclass.postMixInProperties.apply(this, arguments);
 			var messages = dojo.i18n.getLocalization("dojo.widget", "DropdownTimePicker", this.lang);
@@ -50,7 +53,10 @@
 		fillInTemplate: function(){
 			dojo.widget.DropdownTimePicker.superclass.fillInTemplate.apply(this, arguments);
 
-			var timeProps = { widgetContainerId: this.widgetId, lang: this.lang };
+			var timeProps = { 
+				widgetContainerId: this.widgetId, lang: this.lang ,
+				templateCssPath:  this.templateCssPath, templatePath: this.templatePath
+			};
 			this.timePicker = dojo.widget.createWidget("TimePicker", timeProps, this.containerNode, "child");
 			dojo.event.connect(this.timePicker, "onSetTime", this, "onSetTime");
 			dojo.event.connect(this.inputNode,  "onchange",  this, "onInputChange");
Index: src/widget/DropdownDatePicker.js
===================================================================
--- src/widget/DropdownDatePicker.js	(revision 8620)
+++ src/widget/DropdownDatePicker.js	(working copy)
@@ -82,6 +82,12 @@
 		//	disable all incremental controls, must pick a date in the current display
 		staticDisplay: false,
 		
+		//If true disabled the input from keyboard
+		inputNotAllowed:false,
+		
+		templatePath:  dojo.uri.dojoUri("src/widget/templates/DatePicker.html"),
+		templateCssPath:  dojo.uri.dojoUri("src/widget/templates/DatePicker.css"),
+
 		postMixInProperties: function(localProperties, frag){
 			// summary: see dojo.widget.DomWidget
 
@@ -108,7 +114,9 @@
 			//attributes to be passed on to DatePicker
 			var dpArgs = {widgetContainerId: this.widgetId, lang: this.lang, value: this.value,
 				startDate: this.startDate, endDate: this.endDate, displayWeeks: this.displayWeeks,
-				weekStartsOn: this.weekStartsOn, adjustWeeks: this.adjustWeeks, staticDisplay: this.staticDisplay};
+                               weekStartsOn: this.weekStartsOn, adjustWeeks: this.adjustWeeks, staticDisplay: this.staticDisplay, 
+                               templateCssPath:  this.templateCssPath, templatePath: this.templatePath
+                               };
 
 			//build the args for DatePicker based on the public attributes of DropdownDatePicker
 			this.datePicker = dojo.widget.createWidget("DatePicker", dpArgs, this.containerNode, "child");
