--- src/widget/FloatingPane.js	2007-05-15 10:08:02.000000000 +0200
+++ ./FloatingPane.js	2007-05-15 10:07:40.000000000 +0200
@@ -408,6 +408,7 @@
 	{
 		windowState: "minimized",
 		displayCloseAction: true,
+		
 		postCreate: function(){
 			dojo.widget.ModalDialogBase.prototype.postCreate.call(this);
 			dojo.widget.ModalFloatingPane.superclass.postCreate.call(this);
@@ -416,6 +417,8 @@
 			dojo.widget.ModalFloatingPane.superclass.show.apply(this, arguments);
 			this.showModalDialog();
 			this.placeModalDialog();
+			//Allow to input via keyboard
+			this.giveMeFocus();
 			//place the background div under this modal pane
 			this.shared.bg.style.zIndex = this.domNode.style.zIndex-1;
 		},
@@ -427,20 +430,31 @@
 		},
                reOrderWindowsAndBg: function(){
                        var floatingPanes= dojo.widget.manager.getWidgetsByType(this.widgetType);
-                       var windows = [];
-                       for (var x=0; x<floatingPanes.length; x++) {
-                               if (this.widgetId != floatingPanes[x].widgetId) {
-                                               var floating = floatingPanes[x];
-                                               floating.bg.style.zIndex = floating.domNode.style.zIndex-1;
-                               }
-                       }
-               },
+			for (var x=0; x<floatingPanes.length; x++) {
+				if (this.widgetId != floatingPanes[x].widgetId) {
+					var floating = floatingPanes[x];
+					floating.shared.bg.style.zIndex = floating.domNode.style.zIndex-1;
+				}
+			}
+			//disconnect on the hidden window, will be set again in show
+			dojo.event.disconnect(document.documentElement, "onkey", this, "_onKey");
+		},
+		giveMeFocus: function(){
+			//disconnect onkey on all window except me
+			var floatingPanes= dojo.widget.manager.getWidgetsByType(this.widgetType);
+			var windows = [];
+			for (var x=0; x<floatingPanes.length; x++) {
+				if (this.widgetId != floatingPanes[x].widgetId) {
+					dojo.event.disconnect(document.documentElement, "onkey", floatingPanes[x], "_onKey");
+				}
+			}
+		},
 		closeWindow: function(){
 			this.hide();
 			dojo.widget.ModalFloatingPane.superclass.closeWindow.apply(this, arguments);
-               },
-               onMouseDown : function(){
-                       //Do nothing, in that case we DO NOT need to bring the clicked window to the top 
-               }
+		},
+		onMouseDown : function(){
+			//Do nothing, in that case we DO NOT need to bring the clicked window to the top 
+		}
 	}
 );
