Index: src/widget/SplitContainer.js
===================================================================
--- src/widget/SplitContainer.js	(revision 8602)
+++ src/widget/SplitContainer.js	(working copy)
@@ -1,3 +1,4 @@
+
 dojo.provide("dojo.widget.SplitContainer");
 
 //
@@ -154,7 +155,7 @@
 
         // Remove widget and repaint
         dojo.widget.SplitContainer.superclass.removeChild.call(this, widget, arguments);
-        this.onResized();
+		this.onResized();
     },
 
     addChild: function(widget, overrideContainerNode, pos, ref, insertIndex){
@@ -247,6 +248,8 @@
 			this.children[i].checkSize();
 			pos += size;
 		}
+		
+		this.onResize();
 	},
 
 	movePanel: function(panel, pos, size){
@@ -445,6 +448,12 @@
 
 		this.layoutPanels();
 	},
+	
+	/**
+	 * Resize handler for to listen to the resizing
+	 */
+	onResize: function(){
+	},
 
 	showSizingLine: function(){
 
