Index: src/widget/DropdownDatePicker.js
===================================================================
--- src/widget/DropdownDatePicker.js	2007-05-14 16:41:34.000000000 +0200
+++ src/widget/DropdownDatePicker.js	2007-05-14 16:41:48.000000000 +0200
@@ -93,6 +93,9 @@
 		//	disable all incremental controls, must pick a date in the current display
 		staticDisplay: false,
 		
+               //If true disabled the input from keyboard
+               inputNotAllowed:false,
+
                templatePath:  dojo.uri.dojoUri("src/widget/templates/DatePicker.htm"),
                templateCssPath:  dojo.uri.dojoUri("src/widget/templates/DatePicker.css"),
 
@@ -136,6 +139,10 @@
 			}
 			this.containerNode.explodeClassName = "calendarBodyContainer";
 			this.valueNode.name=this.name;
+
+                       //Can disable the input if inputNotAllowed = true
+                       this.inputNode.disabled = this.inputNotAllowed;
+
 		},
 
 		getValue: function(){
@@ -229,6 +236,12 @@
 		destroy: function(/*Boolean*/finalize){
 			this.datePicker.destroy(finalize);
 			dojo.widget.DropdownDatePicker.superclass.destroy.apply(this, arguments);
+               },
+
+               enable: function() {
+                       // summary: enable this widget to accept user input
+                       dojo.widget.DropdownContainer.superclass.enable.apply(this, arguments);
+                       this.inputNode.disabled = this.inputNotAllowed;
 		}
 	}
 );
Index: src/widget/DropdownTimePicker.js
===================================================================
--- src/widget/DropdownTimePicker.js	2007-05-14 16:41:34.000000000 +0200
+++ src/widget/DropdownTimePicker.js	2007-05-14 16:41:48.000000000 +0200
@@ -67,6 +67,9 @@
 		// 	name of the form element, used to create a hidden field by this name for form element submission.
 		name: "",
 
+               //If true disabled the input from keyboard
+               inputNotAllowed:false,
+
                templatePath: dojo.uri.dojoUri("src/widget/templates/TimePicker.htm"),
                templateCssPath: dojo.uri.dojoUri("src/widget/templates/TimePicker.css"),
 
@@ -131,6 +134,8 @@
 			this.containerNode.style.zIndex = this.zIndex;
 			this.containerNode.explodeClassName = "timeContainer";
 			this.valueNode.name = this.name;
+                       //Can disable the input if inputNotAllowed = true
+                       this.inputNode.disabled = this.inputNotAllowed;
 		},
 		
 		getValue: function(){
@@ -232,6 +237,12 @@
 		destroy: function(/*Boolean*/finalize){
 			this.timePicker.destroy(finalize);
 			dojo.widget.DropdownTimePicker.superclass.destroy.apply(this, arguments);
+               },
+
+               enable: function() {
+                       // summary: enable this widget to accept user input
+                       dojo.widget.DropdownContainer.superclass.enable.apply(this, arguments);
+                       this.inputNode.disabled = this.inputNotAllowed;
 		}
 	}
 );