Index: /Users/doume/Dev/Wicket/wicket-contrib-dojo-1.3/src/main/java/wicket/contrib/dojo/dojo-0.4/src/dnd/HtmlDragManager.js
===================================================================
--- /Users/doume/Dev/Wicket/wicket-contrib-dojo-1.3/src/main/java/wicket/contrib/dojo/dojo-0.4/src/dnd/HtmlDragManager.js	(revision 1724)
+++ /Users/doume/Dev/Wicket/wicket-contrib-dojo-1.3/src/main/java/wicket/contrib/dojo/dojo-0.4/src/dnd/HtmlDragManager.js	(working copy)
@@ -329,7 +329,11 @@
 			var tn = tempTarget.domNode;
 			//only cache dropTarget which can accept current dragSource
 			if(!tn || dojo.lang.find(tempTarget.acceptedTypes, this.dragSource.type) < 0){ return; }
-			var abs = dojo.html.getAbsolutePosition(tn, true);
+			var absElt = tn;
+			if (tn.id){
+				absElt = tn.id;
+			}
+			var abs = dojo.html.getAbsolutePosition(absElt, true);
 			var bb = dojo.html.getBorderBox(tn);
 			this.dropTargetDimensions.push([
 				[abs.x, abs.y],	// upper-left
