Index: src/widget/SplitContainer.js
===================================================================
--- src/widget/SplitContainer.js	(revision 8599)
+++ src/widget/SplitContainer.js	(working copy)
@@ -242,8 +242,17 @@
 			this.children[i].position = pos;
 			pos += size;
 		}
+
+               this.onResize();
 	},
 
+
+       /**
+        * Resize handler for to listen to the resizing
+        */
+       onResize: function(){
+       },
+
 	_movePanel: function(/*Widget*/ panel, pos, size){
 		if (this.isHorizontal){
 			panel.domNode.style.left = pos + 'px';
