Index: /Users/doume/Dev/Wicket/wicket-contrib-dojo-1.3/src/main/java/wicket/contrib/dojo/dojo-0.4/src/i18n/calendar/nls/fr/gregorianExtras.js
===================================================================
--- /Users/doume/Dev/Wicket/wicket-contrib-dojo-1.3/src/main/java/wicket/contrib/dojo/dojo-0.4/src/i18n/calendar/nls/fr/gregorianExtras.js	(revision 0)
+++ /Users/doume/Dev/Wicket/wicket-contrib-dojo-1.3/src/main/java/wicket/contrib/dojo/dojo-0.4/src/i18n/calendar/nls/fr/gregorianExtras.js	(revision 0)
@@ -0,0 +1,14 @@
+/*
+	Copyright (c) 2004-2006, The Dojo Foundation
+	All Rights Reserved.
+
+	Licensed under the Academic Free License version 2.1 or above OR the
+	modified BSD license. For more information on Dojo licensing, see:
+
+		http://dojotoolkit.org/community/licensing.shtml
+*/
+
+// Dojo-specific extensions to the CLDR
+({
+	'dateFormat-yearOnly': "yyyy"
+})
\ No newline at end of file
Index: /Users/doume/Dev/Wicket/wicket-contrib-dojo-1.3/src/main/java/wicket/contrib/dojo/dojo-0.4/src/widget/nls/fr/DropdownDatePicker.js
===================================================================
--- /Users/doume/Dev/Wicket/wicket-contrib-dojo-1.3/src/main/java/wicket/contrib/dojo/dojo-0.4/src/widget/nls/fr/DropdownDatePicker.js	(revision 0)
+++ /Users/doume/Dev/Wicket/wicket-contrib-dojo-1.3/src/main/java/wicket/contrib/dojo/dojo-0.4/src/widget/nls/fr/DropdownDatePicker.js	(revision 0)
@@ -0,0 +1,13 @@
+/*
+	Copyright (c) 2004-2006, The Dojo Foundation
+	All Rights Reserved.
+
+	Licensed under the Academic Free License version 2.1 or above OR the
+	modified BSD license. For more information on Dojo licensing, see:
+
+		http://dojotoolkit.org/community/licensing.shtml
+*/
+
+({
+		selectDate: "Choisir une date"
+})
